### ISSUE TITLE
Inquiry on Pre-training Generator

### ISSUE THREAD

**Issue by jangbi1** [LS4GAN/uvcgan2#22]
Hello, @usert5432

First of all, thank you for sharing your outstanding research.

I'm writing to inquire about training on my custom datasets (A & B), each containing approximately 15,000 images with 3x320x320. I am performing an unpaired translation and have some inquiries regarding the process.

I am particularly curious about the necessity of pre-training generator stage. In my dataset, it appears to take around 80 hours per epoch...., and I noticed in a paper that pretraining is recommended. I am wondering if it is necessary, and if so, how many epochs would be advisable for sufficient training? My current setup involves a batch size of 64 and utilizes two RTX 8000 GPUs, taking 30GB memories each.

Additionally, I'm wondering if bypassing the pretraining phase and proceeding directly to translation could be feasible, as it appears to be less time-consuming in benchmarking repository.

I look forward to your response. Thank you once again.

Sincerely,

Jangbi

**Comment by usert5432** [comment]
Hello @jangbi1,

Thank you for your interest in our work.

> I am particularly curious about the necessity of pre-training generator stage. In my dataset, it appears to take around 80 hours per epoch...., and I noticed in a paper that pretraining is recommended.
> Additionally, I'm wondering if bypassing the pretraining phase and proceeding directly to translation could be feasible, as it appears to be less time-consuming in benchmarking repository.

The pretraining is not necessary. The translation usually works without any pretraining. But, pretraining does improve the quality of the translation.

> I am wondering if it is necessary, and if so, how many epochs would be advisable for sufficient training?

I am not sure about how many epochs is necessary for the pretraining. I belive, the number of pre-training iterations can be reduced by a factor of 5 without any loss of the performance. But this is just my intuition, I do not know for sure.

> I am particularly curious about the necessity of pre-training generator stage. In my dataset, it appears to take around 80 hours per epoch

80 hours sounds a bit too much for images of size 320x320. If pre-training is necessary, I would suggest to do the following:
1. `uvcgan2` tries to be smart and automatically uses all the available GPUs. But, sometimes, using many GPUs actually worsens the training speed due to hardware issues. So, one may try binding the pre-training to a single GPU with CUDA_VISIBLE_DEVICES=0 and retrying it with a smaller batch size.
2. You could also try reducing `steps_per_epoch` parameter down to something like 100-200 to reduce the number of pre-training iterations and speed it up.
